src/server/scripts/Events/midsummer.cpp

```diff
@@ -1195,6 +1195,49 @@ class spell_midsummer_torch_catch : public SpellScript
     }
 };
 
+// 46592 - Summon Ahune Lieutenant
+class spell_midsummer_summon_ahune_lieutenant : public SpellScript
+{
+    PrepareSpellScript(spell_midsummer_summon_ahune_lieutenant);
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        uint32 zoneId = caster->GetZoneId();
+        uint32 npcEntry = 0;
+
+        switch (zoneId)
+        {
+        case 331: // Ashenvale
+            npcEntry = 26116;
+            break;
+        case 405: // Desolace
+            npcEntry = 26178;
+            break;
+        case 33: // Stranglethorn Vale
+            npcEntry = 26204;
+            break;
+        case 51: // Searing Gorge
+            npcEntry = 26214;
+            break;
+        case 1377: // Silithus
+            npcEntry = 26215;
+            break;
+        case 3483: // Hellfire Peninsula
+            npcEntry = 26216;
+            break;
```

Comment:
I am really not a fan of all the magic numbers. It's okay that you add a comment to the zone names, but do the same for the npcs pls

Reply:
> I am really not a fan of all the magic numbers.

Nor am I, I promise you. If you'd like, I can replace them, I only did it this way because similar implementations elsewhere also used magic numbers and comments.

> [...] do the same for the npcs

Done.